polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -948,6 +956,115 @@ pub(crate) fn sanitize_bitfields<T: crate::inclusion::Config>(
 	bitfields
 }
 
+/// Perform required checks for given candidate receipt.
+///
+/// Returns `true` if candidate descriptor is version 1.
+///
+/// Otherwise returns `false` if:
+/// - version 2 descriptors are not allowed
+/// - the core index in descriptor doesn't match the one computed from the commitments
+/// - the `SelectCore` signal does not refer to a core at the top of claim queue
+fn sanitize_backed_candidate_v2<T: crate::inclusion::Config>(
+	candidate: &BackedCandidate<T::Hash>,
+	allowed_relay_parents: &AllowedRelayParentsTracker<T::Hash, BlockNumberFor<T>>,
+	allow_v2_receipts: bool,
+) -> bool {
+	if candidate.descriptor().version() == CandidateDescriptorVersion::V1 {
+		return true
+	}
+
+	let Some((core_selector, cq_offset)) = candidate.candidate().commitments.selected_core() else {
+		log::debug!(
+			target: LOG_TARGET,
+			"Dropping V2 candidate receipt {:?} for paraid {:?}, no `SelectCore` commitment.",
+			candidate.candidate().hash(),
+			candidate.descriptor().para_id()
+		);
+
+		return false
+	};
+
+	// Check session index in the receipt. As we drop allowed relay parents at session change
+	// we only allow here candidates that have the session index equal to the current session.
+	let Some(session_index) = candidate.descriptor().session_index() else {
+		log::debug!(
+			target: LOG_TARGET,
+			"Invalid V2 candidate receipt {:?} for paraid {:?}, missing session index.",
+			candidate.candidate().hash(),
+			candidate.descriptor().para_id(),
+		);
+		return false
+	};
+
+	if session_index != shared::CurrentSessionIndex::<T>::get() {
```

Comment:
We still need to accept relay parent from prev session, otherwise nothing can be backed in first block of new session.

Reply:
Ok, this is fine as we the descriptor contains the session index when the candidate is expected to be backed.